Cumulus 853 feature docs

Jkovarik opened the pull request:
**Summary:** Summary of changes

Addresses [CUMULUS-853](https://bugs.earthdata.nasa.gov/browse/CUMULUS-853)

## Changes

This PR adds feature/task documentation related to updates made in CUMULUS-670 for ancillary metadata export to CMR.    It's a seperate PR in addition to the main CUMULUS-853 PR (https://github.com/nasa/cumulus/pull/889) due to: 

* It's not part of the original AC
* It includes functional changes to the organization of the docs page 

PR will be updated to merge into master if the original PR is approved and this one winds up discussed heavily.  

## PR Checklist

- [ X ] Update CHANGELOG
- [ ] Unit tests (n/a)
- [X ] Adhoc testing
- [ ] Integration tests (n/a)

laurenfrederick commented on docs/data-cookbooks/setup.md:
```diff
@@ -47,7 +47,7 @@ The schema for collections can be found [here](https://github.com/nasa/cumulus/t
 |regex|`"^MOD09GQ\\.A[\\d]{7}\\.[\\S]{6}\\.006\\.[\\d]{13}\\.hdf$"`|Yes|Regex used to identify the file|
 |sampleFileName|`MOD09GQ.A2017025.h21v00.006.2017034065104.hdf"`|Yes|Filename used to validate the provided regex|
-|fileType|`"data"`|No|Value to be assigned to the granule file fileType.  CNM file types will be used by Cumulus CMR steps as the fileType, non-CNM values will be treated as the 'data' fileType|
+|fileType|`"data"`|No|Value to be assigned to the Granule File Filetype. CNM filetypes used by Cumulus CMR steps, non-CNM values will be treated as 'data' fileType.  Currently only utilized in DiscoverGranules task|
```
Should we list the CNM filetypes here?

Jkovarik replied:
I think that would probably clutter the table.  I could possibly link the the CNM reference wiki page though.

Jkovarik replied:
I mean more than it already is.... 🤔

laurenfrederick commented on docs/workflow_tasks/move_granules.md:
```diff
@@ -0,0 +1,57 @@
+
+## Summary
+
+This task utilizes the incoming config.input_granules and the task input list of s3 URIs along with the rest of the configuration objects to do the following for a list of files assigned to a single collection:
```
Can we put config.input_granules, etc in code blocks: `config.input_granules`

Jkovarik replied:
👍  e01520ad2

laurenfrederick commented on docs/workflow_tasks/move_granules.md:
```diff
@@ -0,0 +1,57 @@
+
+* Take the list of incoming files and sort them into a list of granule objects.  Assign files to existing granules where appropriate.
+
+* Move granules from their 'staging' location to the final location (as configured )
```
There is an extra space after `as configured`, but I would change this to (as configured in the Sync Granules task)

Jkovarik replied:
62e4f3ca1

laurenfrederick commented on docs/workflow_tasks/move_granules.md:
```diff
@@ -0,0 +1,57 @@
+
+* Update the config.input_granules object with the new file locations.
+
+* If the granule has a ECHO10/UMM CMR file(.cmr.xml or .cmr.json) file included in the config.input_granules, open that file and update access locations, and add it to the appropriate access URL category for the CMR filetype as defined by granule CNM filetype.   Set the CMR file to 'metadata' in the output granules object and add it to  the granule files if it's not already present.
```
I would make this a bulleted list underneath this bullet.

Jkovarik replied:
Good idea!   2da45d0c8

laurenfrederick commented on docs/workflow_tasks/move_granules.md:
```diff
@@ -0,0 +1,57 @@
+
+* If the granule has a ECHO10/UMM CMR file(.cmr.xml or .cmr.json) file included in the config.input_granules, open that file and update access locations, and add it to the appropriate access URL category for the CMR filetype as defined by granule CNM filetype.   Set the CMR file to 'metadata' in the output granules object and add it to  the granule files if it's not already present.
+
+  Please note: **Granules without a valid CNM type set in config.input_granules will be treated as 'data' in the updated CMR metadata file**
```
specify that it's the `fileType` field

Jkovarik replied:
3a5d0c987

laurenfrederick replied:
👍

laurenfrederick commented on docs/workflow_tasks/move_granules.md:
```diff
@@ -0,0 +1,57 @@
+
+## Examples
+
+See [the SIPS workflow cookbook](data-cookbooks/sips-workflow) for an example of this task in a workflow
```
This link doesn't work

Jkovarik replied:
31b89d03a

laurenfrederick replied:
👍

laurenfrederick commented on docs/workflow_tasks/parse_pdr.md:
```diff
@@ -0,0 +1,77 @@
+
+## Examples
+
+See [the SIPS workflow cookbook](data-cookbooks/sips-workflow) for an example of this task in a workflow
```
Link doesn't work

Jkovarik replied:
31b89d03a

laurenfrederick replied:
👍

laurenfrederick commented on docs/features/ancillary_metadata.md:
```diff
@@ -0,0 +1,27 @@
+  Uses a task-specific mapping to map PDR 'FILE_TYPE' to a CNM type to set fileType on granules from the PDR.
+
+### CNMToCMALambdaFunction
+  Natively supports fileTypes that are included in incoming messages to a [CNM Workflow](data-cookbooks/cnm-workflow).
```
Link does not work

Jkovarik replied:
31b89d03a

laurenfrederick replied:
👍

Jkovarik commented on docs/workflow_tasks/discover_granules.md:
```diff
@@ -0,0 +1,55 @@
+
+The purpose of this task is to facilitate ingest of data that does not conform to either a PDR/[SIPS](../data-cookbooks/sips-workflow) discovery mechanism, a [CNM Workflow](../data-cookbooks/sips-workflow) or direct injection of workflow triggering events into Cumulus core components.
+
+The task utilizes a defined [collection](../data-cookbooks/setup#collections) in concert with a defined [provider](data-cookbooks/setup#collections) to scan a location for files matching the defined collection configuration, assemble those files into groupings by granule, and passes the constructed granules object as an output.
```
bae94f802

laurenfrederick replied:
👍

Jkovarik commented on docs/workflow_tasks/discover_granules.md:
```diff
@@ -0,0 +1,55 @@
+
+## Summary
+
+The purpose of this task is to facilitate ingest of data that does not conform to either a PDR/[SIPS](../data-cookbooks/sips-workflow) discovery mechanism, a [CNM Workflow](../data-cookbooks/sips-workflow) or direct injection of workflow triggering events into Cumulus core components.
```
b0660d1df

laurenfrederick replied:
👍

Jkovarik commented:
@laurenfrederick  thanks for the review (and catching the broken links) , I think I've addressed your comments.  

re: diagraming inputs/outputs - I like that idea, I'm just currently unhappy with our setup for generating those docs.  Might be a good IP Sprint task to find a common format we can support.  

Also re: vision, I'm thinking I'm going to enter a ticket to do the work of adding the remaining tasks, and highlight that we don't want to make those entries about documenting *all* schema values or providing a duplicate of the schema interface, more a narrative that helps users interpret workflows and the schemas we provide.

ifestus commented on docs/workflow_tasks/discover_granules.md:
```diff
@@ -0,0 +1,55 @@
+
+### Input
+
+This task does not expect an incoming payload.
```
This may be not worth mentioning, but I think this could be misconstrued as the task expecting _no_ input, which is incorrect. Would it be worthwhile to make the distinction that it doesn't expect any user input, but _does_ expect a Cumulus Message as input?

ifestus replied:
This is definitely implicit because of the configuration discussed after this entry. Completely up to you.

Jkovarik replied:
Part of the problem is we have task input which contains an Input and configuration objects.    I could call it "Incoming Event" as a header instead of Input(s), but that might be misconstrued by users that don't really understand CMA. 

I really don't like either option 🙁

ifestus commented on docs/workflow_tasks/move_granules.md:
```diff
@@ -0,0 +1,60 @@
+id: move_granules
+title: Move Granules
+hide_title: true
+---
```
Just a note, this documentation may have to change once @mennovandiermen 's most recent PR goes through #883

Jkovarik replied:
Right you are.    I'll coordinate with @laurenfrederick  and @mennovandiermen
